src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs

```diff
@@ -375,9 +375,11 @@ public static string GetCamelCasedSuggestion(string name)
             => Regex.Replace(
                 name,
                 @"(?x)
-                    (?<=\p{Lu}) # after any uppercase character,
+                    ((?<=\p{Lu}) # after any uppercase character,
                     \p{Lu}+     # find a series of uppercase characters,
-                    (?=\p{Lu}|\z) # followed by an uppercase character or the end of the string
+                    (?=\p{Lu}|\z)) # followed by an uppercase character or the end of the string
+                    |(\A[A-Z]) # or it happens to be the first character
+                    |((?<=[a-z]){1}[A-Z]\z) # or it is an uppercase at the end, next to a lowercase
```

Comment:
which case is this last one for? example?

Reply:
`ManyThingS` becomes `manyThings`

Reply:
I see, ok, honestly we could also keep it as it, but fine, it's a suggestion, so the user evaluates at the end anyway :)